bench/workgen/workgen.cpp

```diff
@@ -1846,11 +1847,23 @@ ThreadRunner::op_run(Operation *op)
             if (ret == 0)
                 cursor->reset(cursor);
             else {
-                retry_op = true;
+                /*
+                 * We don't retry on a WT_ROLLBACK error when:
+                 * - it is a mirrored operation as Workgen will create a new transaction and
+                 * - the mirror table is the one that faced the WT_ROLLBACK error as the operation
+                 * on the base table will be lost.
+                 */
+                if (op->_random_table && _icontext->_dyn_table_runtime[tint].has_mirror() &&
```

Comment:
So we always operate on the base first? If we may operate on the mirror first and we rollback on the base, we will lose the operation on the mirror.

Reply:
Yes, I checked the code and when a table is selected with mirroring, we always select the base.

Cancelling the operation whether we are dealing with the base or the mirror would be the safest approach though and more robust long term if the first statement is no longer true.